packages/server/src/course/course.controller.ts

```diff
@@ -525,64 +529,87 @@ export class CourseController {
     await this.courseService.removeUserFromCourse(userCourse);
   }
 
-  /* TODO: This is the old login registration form.
-  @Post('submit_course')
-  async submitCourse(@Body() body: SubmitCourseParams): Promise<void> {
-    if (body.password !== process.env.APPLY_PASSWORD) {
-      throw new UnauthorizedException(
-        ERROR_MESSAGES.courseController.invalidApplyURL,
-      );
-    }
-
-    const season = body.semester.split(' ')[0];
-    const year = parseInt(body.semester.split(' ')[1]);
-
-    const semester = await SemesterModel.findOne({
-      where: { season, year },
+  @Post('register_courses')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async registerCourses(
+    @Body() body: RegisterCourseParams[],
+    @UserId() userId: number,
+  ): Promise<void> {
+    // obtains the ProfSectionGroupsModel of the professor
+    const profSectionGroups = await ProfSectionGroupsModel.findOne({
+      where: { profId: userId },
     });
-    if (!semester)
-      throw new BadRequestException(
-        ERROR_MESSAGES.courseController.noSemesterFound,
+
+    // iterate over each section group registration
+    for (const courseParams of body) {
+      // finds professor's section group with matching name
+      const sectionGroup = profSectionGroups.sectionGroups.find(
+        (sg) => sg.name === courseParams.name,
       );
 
-    let course = null;
-    try {
-      // create the submitted course
-      course = await CourseModel.create({
-        name: body.name,
-        coordinator_email: body.coordinator_email,
-        icalURL: body.icalURL,
-        semesterId: semester.id,
-        enabled: false,
-        pending: true,
-        timezone: body.timezone,
-      }).save();
-    } catch (err) {
-      console.error(err);
-      throw new HttpException(
-        ERROR_MESSAGES.courseController.createCourse,
-        HttpStatus.INTERNAL_SERVER_ERROR,
+      const khourySemesterParsed = this.loginCourseService.parseKhourySemester(
+        sectionGroup.semester,
       );
-    }
+      const semester = await SemesterModel.findOne({
+        where: {
+          season: khourySemesterParsed.season,
+          year: khourySemesterParsed.year,
+        },
+      });
 
-    try {
-      // create CourseSectionMappings for each unique submitted section number
-      new Set(body.sections).forEach(async (section) => {
-        await CourseSectionMappingModel.create({
-          genericCourseName: body.name,
-          section,
-          courseId: course.id,
+      let course = null;
+      try {
+        // create the submitted course
+        course = await CourseModel.create({
+          name: courseParams.displayName,
+          sectionGroupName: courseParams.name,
+          coordinator_email: courseParams.coordinator_email,
+          icalURL: courseParams.iCalURL,
+          semesterId: semester.id,
+          enabled: true,
+          pending: false,
```

Comment:
@nicoledanuwidjaja id wait to merge the pending field PR, bc you will have to delete this line anyways